api/kyverno/v1/common_types.go

```diff
@@ -243,12 +243,6 @@ type Mutation struct {
 	ForEachMutation []ForEachMutation `json:"foreach,omitempty" yaml:"foreach,omitempty"`
 }
 
-type TargetMutation struct {
```

Comment:
I created this separate struct to be able to extend it if needed. 

For example, if a user wants to define conditions on the target resource, we can add an additional `preconditions` block to this struct.

I would like to see if there's any use case like this once the feature is released. For now we can remove it.